libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/AppLockView/UIColor+SFSDKPasscodeView.m

```diff
@@ -30,15 +30,74 @@
 @implementation UIColor (SFSDKPasscodeView)
 
 + (UIColor *)passcodeViewBackgroundColor {
-    return [UIColor colorWithRed:245.0f/255.0f green:246.0f/255.0f blue:250.0f/255.0f alpha:1.0f];
+    UIColor *lightStyleColor = [UIColor colorWithRed:245.0f/255.0f green:246.0f/255.0f blue:250.0f/255.0f alpha:1.0f];
+    if (@available(iOS 13.0, *)) {
+        return [[UIColor alloc] initWithDynamicProvider:^UIColor * _Nonnull(UITraitCollection * _Nonnull traitCollection) {
+            if (traitCollection.userInterfaceStyle == UIUserInterfaceStyleDark) {
+                return [UIColor colorWithRed:8.0f/255.0f green:7.0f/255.0f blue:7.0f/255.0f alpha:1.0f];
+            } else {
+                return lightStyleColor;
+            }
+        }];
+    }
+    return lightStyleColor;
+}
+
++ (UIColor *)passcodeViewSecondaryBackgroundColor {
+    UIColor *lightStyleColor = [UIColor whiteColor];
+    if (@available(iOS 13.0, *)) {
+        return [[UIColor alloc] initWithDynamicProvider:^UIColor * _Nonnull(UITraitCollection * _Nonnull traitCollection) {
+            if (traitCollection.userInterfaceStyle == UIUserInterfaceStyleDark) {
+                return [UIColor colorWithRed:43.0f/255.0f green:40.0f/255.0f blue:38.0f/255.0f alpha:1.0f];
+            } else {
+                return lightStyleColor;
+            }
+        }];
+    }
+    return lightStyleColor;
 }
 
 + (UIColor *)passcodeViewTextColor {
-    return [UIColor colorWithRed:22.0f/255.0f green:50.0f/255.0f blue:92.0f/255.0f alpha:1.0f];
+    UIColor *lightStyleColor = [UIColor colorWithRed:22.0f/255.0f green:50.0f/255.0f blue:92.0f/255.0f alpha:1.0f];
+    if (@available(iOS 13.0, *)) {
+        return [[UIColor alloc] initWithDynamicProvider:^UIColor * _Nonnull(UITraitCollection * _Nonnull traitCollection) {
+            if (traitCollection.userInterfaceStyle == UIUserInterfaceStyleDark) {
+                return [UIColor colorWithRed:240.0f/255.0f green:240.0f/255.0f blue:240.0f/255.0f alpha:1.0f];
+            } else {
+                return lightStyleColor;
+            }
+        }];
+    }
+    return lightStyleColor;
 }
 
 + (UIColor *)passcodeViewBorderColor {
-    return [UIColor colorWithRed:217.0f/255.0f green:221.0f/255.0f blue:230.0f/255.0f alpha:1.0f];
+    UIColor *lightStyleColor = [UIColor colorWithRed:217.0f/255.0f green:221.0f/255.0f blue:230.0f/255.0f alpha:1.0f];
+    if (@available(iOS 13.0, *)) {
+        return [[UIColor alloc] initWithDynamicProvider:^UIColor * _Nonnull(UITraitCollection * _Nonnull traitCollection) {
+            if (traitCollection.userInterfaceStyle == UIUserInterfaceStyleDark) {
+                return [UIColor colorWithRed:62.0f/255.0f green:62.0f/255.0f blue:60.0f/255.0f alpha:1.0f];
```

Comment:
Can this be derived from the light color with a formula?
If that's the case, we could have a helper method:
```objective-c
+ (UIColor*) darkColorIfNeeded:(UIColor*)lightStyleColor {
    UIColor* color = lightStyleColor;
    if (@available(iOS 13.0, *)) {
        if (traitCollection.userInterfaceStyle == UIUserInterfaceStyleDark) {
            color = /* calculate dark version of lightStyleColor */
        }
    }
    return color;
}
```
Then all those methods would be simplified e.g.
```objective-c
+ (UIColor *)passcodeViewBorderColor {
    return [UIColor darkColorIfNeeded:[UIColor colorWithRed:217.0f/255.0f green:221.0f/255.0f blue:230.0f/255.0f alpha:1.0f]];
}
```

Reply:
I'm not sure about a formula because the colors might change soon anyway (this was a v1 from UX), but maybe something like this so more of the logic could be combined? Better naming suggestions welcome :P

```
+ (UIColor *)passcodeViewNavBarColor {
    UIColor *lightStyleColor = [UIColor whiteColor];
    UIColor *darkStyleColor = [UIColor colorWithRed:43.0f/255.0f green:40.0f/255.0f blue:38.0f/255.0f alpha:1.0f];
    return [UIColor colorForLightStyle:lightStyleColor darkStyle:darkStyleColor];
```

```
+ (UIColor *)colorForLightStyle:(UIColor *)lightStyleColor darkStyle:(UIColor *)darkStyleColor {
    if (@available(iOS 13.0, *)) {
        return [[UIColor alloc] initWithDynamicProvider:^UIColor * _Nonnull(UITraitCollection * _Nonnull traitCollection) {
            if (traitCollection.userInterfaceStyle == UIUserInterfaceStyleDark) {
                return darkStyleColor;
            } else {
                return lightStyleColor;
            }
        }];
    }
    return lightStyleColor;
}
```

Reply:
👍